tracking_manager/models/ir_model.py

```diff
@@ -0,0 +1,206 @@
+                ("model_id.active_custom_tracking", "=", True),
+                ("ttype", "=", "one2many"),
+            ]
+        )
+        related_models = self.env["ir.model"].search(
+            [
+                ("model", "in", fields.mapped("relation")),
+            ]
+        )
+        custom_tracked_fields = self._get_custom_tracked_fields_per_model()
+        res = {}
+        for model in related_models:
+            if model.model not in self.env:
+                # If the model do not exist skip it (ex: during module update)
+                continue
+            if model.model in custom_tracked_fields:
+                tracked_fields = custom_tracked_fields[model.model]
+            else:
+                tracked_fields = model.field_id.filtered(
+                    lambda s, model=model: not s.readonly
+                    and not s.related
+                    and not s.ttype == "one2many"
+                    and s.name in self.env[model.model]._fields
+                ).mapped("name")
+            res[model.model] = {"fields": tracked_fields, "notify": []}
+
+        for field in fields:
+            model_name = field.model_id.model
+            if (
+                model_name in self.env
+                and self.env[model_name]._fields.get(field.name)
+                and field.relation in res
+            ):
+                res[field.relation]["notify"].append(
+                    [self.env[model_name]._fields[field.name].inverse_name, field.name]
+                )
+        return res
+
+    @api.depends("active_custom_tracking")
+    def _compute_automatic_custom_tracking(self):
+        for record in self:
+            record.automatic_custom_tracking = False
+
+    def _default_automatic_custom_tracking_domain_rules(self):
+        return {
+            "product.product": [
+                ("readonly", "=", False),
+                "|",
+                ("ttype", "!=", "one2many"),
+                ("name", "in", ["barcode_ids"]),
+            ],
+            "sale.order": [
+                ("readonly", "=", False),
+                "|",
+                ("ttype", "!=", "one2many"),
+                ("name", "in", ["order_line"]),
+            ],
+            "account.move": [
+                ("readonly", "=", False),
+                "|",
+                ("ttype", "!=", "one2many"),
+                ("name", "in", ["invoice_line_ids"]),
+            ],
+            "default_automatic_rule": [
+                ("ttype", "!=", "one2many"),
+                ("readonly", "=", False),
+            ],
+        }
+
+    @api.depends("automatic_custom_tracking")
+    def _compute_automatic_custom_tracking_domain(self):
+        rules = self._default_automatic_custom_tracking_domain_rules()
+        for record in self:
+            record.automatic_custom_tracking_domain = str(
+                rules.get(record.model) or rules.get("default_automatic_rule")
+            )
```

Comment:
```suggestion
        for record in self:
            automatic_custom_tracking_domain = rules.get(record.model) or rules.get("default_automatic_rule", [])
            automatic_custom_tracking_domain.append(('model', '=', record.model))
            record.automatic_custom_tracking_domain = str(automatic_custom_tracking_domain)
```
would it make sense to do like so considering also the widget='domain' changes I have suggested below?
I believe doing it like that we can improve the UI when adding the domain from the Model view. WDYT?